### PR TITLE
Fix symbolic reference handling in getReferences

### PR DESCRIPTION
**Pull request by billt2006**
I also updated the example in `general.js` so that it would return symbolic references as expected.

### CODE REVIEW

File: lib/repository.js [hunk 1]
```diff
@@ -145,7 +145,13 @@ Repository.prototype.getReference = function(name, callback) {
   }, callback);
 };
 
-Repository.getReferences = function(repo, type, refNamesOnly, callback) {
+Repository.getReferences =
+function(repo, type, refNamesOnly, includeSymbolic, callback) {
+  if (typeof callback === "undefined" &&
```
Comment (tbranyen) [inline lib/repository.js]: Do we need a typeof check here? That's really only necessary if the variable is possibly undeclared.  Simply do `!callback` or `callback == null`.

Comment (billt2006) [inline lib/repository.js]: Alright, I have updated to just use `!callback`.

Comment (tbranyen) [inline lib/repository.js]: Awesome, I'm being picky here as someone could pass a falsy value that isn't undefined and that would break your check.

Comment (billt2006) [inline lib/repository.js]: No problem! You have every right to be picky. :wink:

**Comment by johnhaley81** [comment]
:+1: looks good. I'm going to merge this.

**Pull request merged**